raiden/network/proxies/user_deposit.py

```diff
@@ -360,6 +386,61 @@ def approve_and_deposit(
 
             self._deposit_check_result(transaction_sent, token, total_deposit, amount_to_deposit)
 
+    def plan_withdraw(
+        self, amount: TokenAmount, given_block_identifier: BlockIdentifier
+    ) -> BlockNumber:
+        """ Announce that you plan to withdraw tokens from the UserDeposit contract
+
+        Returns the block number at which the withdraw is ready.
+        """
+        self._plan_withdraw_preconditions(amount, given_block_identifier)
+
+        # Simplify our lives by disallowing concurrent plan_withdraw / withdraw calls
+        with self._withdraw_lock:
```

Comment:
This does not work with restarts. Is that intentional?

Reply:
I didn't want to spend too much effort since the withdraw is currently not used in Raiden itself. But you're right this should be documented at the least.